tests/test_meetings.py

```diff
@@ -0,0 +1,663 @@
+def test_upload_logo_to_theme(meetings):
+    stub(
+        responses.GET,
+        'https://api-eu.vonage.com/beta/meetings/themes/logos-upload-urls',
+        fixture_path='meetings/list_logo_upload_urls.json',
+    )
+    stub(
+        responses.POST,
+        'https://s3.amazonaws.com/roomservice-whitelabel-logos-prod',
+        fixture_path='meetings/null.json',
+        status_code=204,
+    )
+    stub_bytes(
+        responses.PUT,
+        'https://api-eu.vonage.com/beta/meetings/themes/90a21428-b74a-4221-adc3-783935d654db/finalizeLogos',
+        body=b'OK',
+    )
+
+    response = meetings.upload_logo_to_theme(
+        theme_id='90a21428-b74a-4221-adc3-783935d654db',
+        path_to_image='tests/data/meetings/transparent_logo.png',
+        logo_type='white',
+    )
+    assert response == 'Logo upload to theme: 90a21428-b74a-4221-adc3-783935d654db was successful.'
+
+
+@responses.activate
+def test_get_logo_upload_url(meetings):
+    stub(
+        responses.GET,
+        'https://api-eu.vonage.com/beta/meetings/themes/logos-upload-urls',
+        fixture_path='meetings/list_logo_upload_urls.json',
+    )
+
+    url_w = meetings._get_logo_upload_url('white')
+    assert url_w['url'] == 'https://s3.amazonaws.com/roomservice-whitelabel-logos-prod'
+    assert url_w['fields']['X-Amz-Credential'] == 'some-credential'
+    assert url_w['fields']['key'] == 'auto-expiring-temp/logos/white/d92b31ae-fbf1-4709-a729-c0fa75368c25'
+    assert url_w['fields']['logoType'] == 'white'
+    url_c = meetings._get_logo_upload_url('colored')
+    assert url_c['fields']['key'] == 'auto-expiring-temp/logos/colored/c4e00bac-781b-4bf0-bd5f-b9ff2cbc1b6c'
+    assert url_c['fields']['logoType'] == 'colored'
+    url_f = meetings._get_logo_upload_url('favicon')
+    assert url_f['fields']['key'] == 'auto-expiring-temp/logos/favicon/d7a81477-38f7-460c-b51f-1462b8426df5'
+    assert url_f['fields']['logoType'] == 'favicon'
+
+    with pytest.raises(MeetingsError) as err:
+        meetings._get_logo_upload_url('not-a-valid-option')
+    assert str(err.value) == 'Cannot find the upload URL for the specified logo type.'
+
+
+@responses.activate
+def test_upload_to_aws(meetings):
+    stub(
+        responses.POST,
+        'https://s3.amazonaws.com/roomservice-whitelabel-logos-prod',
+        fixture_path='meetings/null.json',
+        status_code=204,
+    )
+
+    with open('tests/data/meetings/list_logo_upload_urls.json') as file:
+        urls = json.load(file)
+    params = urls[0]
+    meetings._upload_to_aws(params, 'tests/data/meetings/transparent_logo.png')
+
+
+@responses.activate
+def test_upload_to_aws_error(meetings):
+    stub(
+        responses.POST,
+        'https://s3.amazonaws.com/not-a-valid-url',
+        status_code=403,
+    )
+
+    with open('tests/data/meetings/list_logo_upload_urls.json') as file:
+        urls = json.load(file)
+
+    params = urls[0]
+    params['url'] = 'https://s3.amazonaws.com/not-a-valid-url'
```

Comment:
AWS will return XML when there is an error

Reply:
Have added an XML mock for this case